hudi-common/src/main/java/org/apache/hudi/common/model/HoodieAvroRecord.java

```diff
@@ -157,7 +157,7 @@ public HoodieRecord truncateRecordKey(Schema recordSchema, Properties props, Str
 
   @Override
   public boolean isDelete(Schema recordSchema, Properties props) throws IOException {
-    if (this.data instanceof BaseAvroPayload) {
+    if (this.data instanceof BaseAvroPayload && !(this.data instanceof AWSDmsAvroPayload)) {
```

Comment:
we should avoid making base class dependent on subclass. Besides, the issue rooted at custom delete marker, which `DefaultHoodieRecordPayload` uses and gets affected by this too. So this won't resolve for `DefaultHoodieRecordPayload` and subclasses